src/MarketingPageAcceptanceTests.Actions/Pages/ContactDetails.cs

```diff
@@ -16,27 +16,45 @@ public bool PageDisplayed()
             return new Common(driver).PageTitleEquals("Contact details");
         }
 
-        public void EnterAllData(IContactDetail firstContact, IContactDetail secondContact = null)
+        public void EnterAllData(IContactDetail firstContact, IContactDetail secondContact = null, bool clearFirst = false)
         {
-            FirstContactComplete(firstContact);
+            FirstContactComplete(firstContact, clearFirst);
 
             if(!(secondContact is null))
             {
-                SecondContactComplete(secondContact);
+                SecondContactComplete(secondContact, clearFirst);
             }
         }
 
-        private void FirstContactComplete(IContactDetail contact)
+        private void FirstContactComplete(IContactDetail contact, bool clearFirst = false)
```

Comment:
You're directly implementing this in the calling method, do you need to set a default value here?